File: src/main/java/no/nav/syfo/controller/internad/DiskresjonskodeADController.java
package no.nav.syfo.controller.internad;

import no.nav.security.spring.oidc.validation.api.ProtectedWithClaims;
import no.nav.syfo.services.DiskresjonskodeService;
import org.springframework.web.bind.annotation.*;

import javax.inject.Inject;

import static no.nav.syfo.oidc.OIDCIssuer.AZURE;
import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;

@RestController
@ProtectedWithClaims(issuer = AZURE)
@RequestMapping(value = "/api/internad/diskresjonskode/{fnr}")
public class DiskresjonskodeADController {

    private DiskresjonskodeService diskresjonskodeService;

    @Inject
    public DiskresjonskodeADController(
            final DiskresjonskodeService diskresjonskodeService
    ) {
        this.diskresjonskodeService = diskresjonskodeService;
    }

    @GetMapping(produces = APPLICATION_JSON_VALUE)
    public DiskresjonskodeSvar getDiskresjonskode(@PathVariable String fnr) {
        return new DiskresjonskodeSvar(diskresjonskodeService.diskresjonskode(fnr));
    }

    public class DiskresjonskodeSvar {
        public String diskresjonskode;

        public DiskresjonskodeSvar(String diskresjonskode) {
            this.diskresjonskode = diskresjonskode;
        }
    }
}

File: src/main/kotlin/no/nav/syfo/controller/DiskresjonskodeADController.kt
package no.nav.syfo.controller

import no.nav.security.spring.oidc.validation.api.ProtectedWithClaims
import no.nav.syfo.oidc.OIDCIssuer.AZURE
import no.nav.syfo.services.DiskresjonskodeService
import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
import org.springframework.web.bind.annotation.*
import javax.inject.Inject

@RestController
@ProtectedWithClaims(issuer = AZURE)
@RequestMapping(value = ["/api/internad/diskresjonskode/{fnr}"])
class DiskresjonskodeADController @Inject
constructor(
        private var diskresjonskodeService: DiskresjonskodeService
) {

    @GetMapping(produces = [APPLICATION_JSON_VALUE])
    fun getDiskresjonskode(@PathVariable fnr: String): DiskresjonskodeSvar {
        return DiskresjonskodeSvar(diskresjonskodeService.diskresjonskode(fnr))
    }

    inner class DiskresjonskodeSvar(var diskresjonskode: String)
}
